app/controllers/admin/edition_workflow_controller.rb

```diff
@@ -98,16 +98,20 @@ def force_publish
   end
 
   def confirm_unpublish
-    @unpublishing = @edition.build_unpublishing(unpublishing_reason_id: UnpublishingReason::Withdrawn.id)
+    @unpublishing = @edition.build_unpublishing
+    @previous_withdrawals = @edition.document.withdrawals
```

Comment:
I'm a bit worried that this is boundless. Do we think it's ok for someone to flip back multiple withdrawals?

As a way to try not make this to big an issue I'd perhaps suggest a compromise of a limit of 5. I can't really imagine why anyone would need more than 1 or 2 or why there'd be a lot - but we've seen the dangers of limitless queries.

Also should it have an ordering? I'm guessing it's relying on implicit ordering at the mo

Reply:
Having looked across Whitehall, the highest number of previous withdrawals on a document currently is 4 or 5 (can't remember off the top of my head) so I'm not too worried about this scenario.

Originally we'd decided just to go for _the most recent_ previous withdrawal (singular) but realised that this would be problematic if publishers accidentally withdraw as 'today' and then need to undo it and reuse an older withdrawal instead. So we allowed for multiple to support that use case.

Reply:
Sure, I think if we're not worried about the scenario just stick an arbitrary limit on it to make it not boundless. Then we can hear if users have a problem where they expected more choice rather than risk a crash one day.

I think you missed the question about ordering btw.